cmake/external/onnxruntime_external_deps.cmake

```diff
@@ -369,7 +369,20 @@ include(eigen)
 include(wil)
 
 if (NOT onnxruntime_MINIMAL_BUILD)
+  if (onnxruntime_PREFER_SYSTEM_LIB)
+    find_package(ONNX CONFIG)
```

Comment:
By default onnxruntime_fetchcontent_makeavailable will call find_package if a package name was provided when the component was declared.

Reply:
Thanks for the suggestion, didn't know that before.
But here's a little bit different, we want to search for ONNXConfig before FindONNX (available in many places) to avoid interception, but cmake don't have a local option for that (there's a global one, but would break common assumptions).
The one in onnxruntime_fetchcontent_makeavailable seems to search module before config.

Reply:
I usually implement it as:
```
find_package(XXX CONFIG)
if (NOT XXX_FOUND)
    find_package(XXX MODULE)
endif ()
if (...)
```
But because ort doesn't have FindONNX we can simply skip the second one.

Reply:
> FetchContent redirection mode
New in version 3.24: A call to find_package() can be redirected internally to a package provided by the [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html#module:FetchContent) module. To the caller, the behavior will appear similar to Config mode, except that the search logic is by-passed and the component information is not used. See [FetchContent_Declare()](https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_declare) and [FetchContent_MakeAvailable()](https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_makeavailable) for further details.

FYI, found a new feature in CMake, in case you're interested.

Reply:
It only supports CONFIG mode.

Reply:
Maybe I'm confused. Could you clarify? I didn't see any CONFIG keyword in the find_package call there